homeassistant/components/transmission/sensor.py

```diff
@@ -122,7 +116,7 @@ def native_value(self) -> float:
         data = self.coordinator.data
         return (
             float(data.download_speed)
-            if self._key == "download"
+            if self.entity_description.key == "download"
```

Comment:
I recommend following up with adding a value access and formatting function to the entity description to avoid the condition in the entity.

See `value_fn` in this example:
https://developers.home-assistant.io/docs/core/entity#example